packages/database/src/migrations/20210817064143-AddPacmossiTotalNumberOfMosquitosByMonthStackedBarChart-modifies-data.js

```diff
@@ -0,0 +1,140 @@
+'use strict';
+
+import { insertObject, generateId, findSingleRecord, findSingleRecordBySql } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const CODE = 'PMOS_Total_Mosquitos_Per_Month';
+const COUNTRIES = ['FJ', 'KI', 'NR', 'NU', 'PG', 'SB', 'TV', 'TO', 'VU'];
+const getDashboardCode = country => `${country}_PacMOSSI_Vector_Surveillance`;
+
+const REPORT_CONFIG = {
+  fetch: {
+    dataElements: ['PMOS_Anopheles', 'PMOS_Aedes', 'PMOS_Culex', 'PMOS_Mansonia'],
+    aggregations: [
+      {
+        type: 'SUM_PER_PERIOD_PER_ORG_GROUP',
+        config: { dataSourceEntityType: 'field_station', aggregationEntityType: 'requested' },
+      },
+    ],
+  },
+  transform: [
+    'keyValueByDataElementName',
+    {
+      transform: 'aggregate',
+      period: 'group',
+      organisationUnit: 'drop',
+      '...': 'sum',
+    },
+    {
+      transform: 'select',
+      "'Anopheles'":
+        'divide($row.PMOS_Anopheles, sum([$row.PMOS_Anopheles, $row.PMOS_Aedes, $row.PMOS_Culex, $row.PMOS_Mansonia]))',
```

Comment:
I would add a 'total' at the beginning and delete it at the end of the transform, but it is up to you. :)

Reply:
Good call, will do!